### ISSUE TITLE
What is the "NIL Provided" used for?

### ISSUE THREAD

**Issue by MasterEnderman** [Ender-Development/OCID#5]
![image](https://github.com/Ender-Development/OCID/assets/24524662/9239434b-f6f4-4358-96d1-119768317e37)
to me is looks like another boolean that can be toggled via `component.computer_reader.provide(0)`

**Comment by Klebestreifen** [comment]
"Nil Provided" is true if `.provide(nil)` or `.provide()` is called. Otherwise something is provided and it is false

When the module is initialized, if nothing is provided yet, the value is therefore always true.

**Comment by MasterEnderman** [comment]
Do you see any usecase for this that i didn't came across yet?

**Comment by Klebestreifen** [comment]
Yes, for troubleshooting and debugging. All aspects always have a value, even if none is provided. This allows you to determine that they are invalid.
